packages/block-library/src/navigation-link/edit.js

```diff
@@ -483,18 +462,17 @@ export default compose( [
 			hasSelectedInnerBlock,
 			getBlockParentsByBlockName,
 			getSelectedBlockClientId,
-			getSettings,
 		} = select( 'core/block-editor' );
 		const { clientId } = ownProps;
 		const rootBlock = head(
 			getBlockParentsByBlockName( clientId, 'core/navigation' )
 		);
 		const navigationBlockAttributes = getBlockAttributes( rootBlock );
```

Comment:
Like the PR for Page List the other day, it should be possible to simplify this now to use the block `context` prop.

Can be left to a separate PR, but I think it'd be good to get to.

Reply:
I'd rather do it as a separate PR together with converting the function to only using hooks, as we currently have an unseemly mix of hooks and HOCs in there 😅